src/main/java/org/wildfly/security/auth/spi/ModifiableAttributesEntry.java

```diff
@@ -18,15 +18,13 @@
 
 package org.wildfly.security.auth.spi;
 
-import java.util.AbstractList;
-
 /**
  * An implementation of {@link Attributes.Entry} which can be used by implementations of {@link Attributes}.  Operations
  * are implemented in terms of methods on {@code Attributes} which do not rely upon entries.
  *
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
  */
-public class SimpleAttributesEntry extends AbstractList<String> implements Attributes.Entry {
+public class ModifiableAttributesEntry extends ReadOnlyAttributesEntry {
```

Comment:
This change, and the ReadOnly variation, are not needed.  Due to the nature of the implementation of this class, if the backing Attributes map is read-only, then the Entry will also be read-only.